docs/CNVs/CNV-methods.tex

```diff
@@ -868,6 +869,98 @@ \subsection{Generative Model for Read Counts} \label{coverage-model}
 \end{equation}
 and thus the ratio $n_{st} / \left[ c_{st} d_s \exp \left( \vW_t E[\vz_s] + m_t \right) \right]$ (with error bars of size $1/\sqrt{n}$ if we want to do a weighted regression) is an estimate of $f_s({\rm GC}_t)$ that we can feed into our favorite regression model.  This is more sophisticated than the standard approach of simply regressing $n$ versus ${\rm GC}$ in that it seeks to explain with GC only the bias that cannot be explained by linear latent features.
 
+
+\subsection{HMM-based segmentation of somatic CNVs} \label{HMM-segmentation}
+In CNV segmentation, the hidden states are defined at genomic loci -- targets or SNPs.  The transition matrix of an HMM for segmentation will be a function of the distances between consecutive loci, similar to what already exists in our germline code.  The emission likelihoods of the HMM will be given by probabilistic models for allele counts at het sites (which we already have) and total coverage of targets (another proposed method).  Broadly speaking, all that remains is to define the hidden states and to tweak the transition model for the somatic case.  For concreteness we will begin the discussion in terms of het allele fraction segmentation.
+
+We already have a probabilistic model $P(a_i, r_i | f_i)$, where $a_i$ and $r_i$ are observed alt and ref allele counts at het site $i$ and $f_i$ is the underlying minor allele fraction of the segment to which this het belongs.  This is the emission likelihood of our HMM.  The hidden states are therefore are discrete set of minor allele fractions $f$.  The discreteness of these states follows from the discreteness of the tumor's phylogeny.  In order to learn a finite set of minor allele fractions without knowing its size a priori we will choose some sufficiently large number (i.e. an overestimate of the number of different values of $f$) $K$ combined with a sparsity-promoting Dirichlet prior on the multinomial weights for the different values of $f$.  A simple idea for the transition matrix would be to assume some ``memory-loss'' scale $D$, so that the probability of forgetting the current CNV state over $d$ bases is $e^{-d/D}$.  If memory is lost, a new CNV state is chosen from a multinomial distribution ${\bf \pi}$.  Thus every site $n$ has an associated memory-loss binary indicator $s_n$ with prior probability $P(s_n = 0) = e^{-d_n/D}$, where $d_n$ is the distance from the previous het to the $n$th het.  This corresponds to the following transition probabilities for the $n$th site:
```

Comment:
should this read "the probability of forgetting the current CNV state over $d$ bases is $1 - e^{-d/D}$", or equivalently "the probability of remembering the current CNV state over $d$ bases is $e^{-d/D}$"?  looking at the code now it looks like you go with `pRemember = Math.exp(-distance / memoryLength)`.  also looks like you have it correct in the javadoc for `AlleleFractionHiddenMarkovModel`.

i might just dispense with the "memory-loss" terminology and go with "transition distance scale" or something along those lines.